salt/modules/git.py

```diff
@@ -273,8 +294,10 @@ def _git_run(command, cwd=None, user=None, password=None, identity=None,
                     redirect_stderr=redirect_stderr,
                     **kwargs)
             finally:
-                if not salt.utils.platform.is_windows() and 'GIT_SSH' in env:
-                    os.remove(env['GIT_SSH'])
+                # Cleanup the temporary ssh wrapper file
+                if tmp_ssh_wrapper and os.path.exists(tmp_ssh_wrapper):
+                    log.debug('Removing ssh wrapper file {0}'.format(tmp_ssh_wrapper))
+                    __salt__['file.remove'](tmp_ssh_wrapper)
```

Comment:
EAFP here. We can skip stat-ing the file by doing it like this:

```python
try:
    __salt__['file.remove'](tmp_ssh_wrapper)
    log.debug('Removed ssh wrapper file %s', tmp_ssh_wrapper)
except AttributeError:
    # No wrapper was used
    pass
except (SaltInvocationError, CommandExecutionError) as exc:
    log.warning('Failed to remove ssh wrapper file %s: %s', tmp_ssh_wraper, exc)
```

Reply:
I can't say I like calling a function with an obviously wrong parameter. I would have kept at least a check `if tmp_ssh_wrapper` which does not require any syscall and would avoid a function call with consequent exception catch but I am happy to oblige.

Note that I wrote it this way because that's exactly how `tmp_identity_file` is removed a couple of lines below. Should I change that too?

Reply:
>I can't say I like calling a function with an obviously wrong parameter.

That's what exception handling is for.

https://en.wikipedia.org/wiki/Python_syntax_and_semantics#Exceptions